src/fm/cp_dlaf_utils.c

```diff
@@ -0,0 +1,42 @@
+/*----------------------------------------------------------------------------*/
+/*  CP2K: A general program to perform molecular dynamics simulations         */
+/*  Copyright 2000-2023 CP2K developers group <https://cp2k.org>              */
+/*                                                                            */
+/*  SPDX-License-Identifier: GPL-2.0-or-later                                 */
+/*----------------------------------------------------------------------------*/
+
+#if defined(__DLAF)
+
+#include <dlaf_c/eigensolver/eigensolver.h>
+#include <dlaf_c/factorization/cholesky.h>
+#include <dlaf_c/init.h>
+
+void dlaf_init() {
+  const char *pika_argv[] = {"cp2k"};
+  const char *dlaf_argv[] = {"cp2k"};
+  dlaf_initialize(1, pika_argv, 1, dlaf_argv);
+}
+
+// Eigensolver (double)
+void dlaf_pdsyevd_wrapper(int n, double *a, int ia, int ja, int desca[9],
+                          double *w, double *z, int iz, int jz, int descz[9],
+                          int *info) {
+  char uplo = 'L'; // Only lower triangular matrices are supported
+  dlaf_pdsyevd(uplo, n, a, ia, ja, desca, w, z, iz, jz, descz, info);
+}
+
+// Cholesky decomposition (double)
+// Wrapper without uplo parameter, avoids passing characters
```

Comment:
Strings (`char*`) yes, but I was having issues with passing single characters (I'm not a Fortran expert and all the information I could find was for strings...). The DLAF C API requires a single character, not a string.

Reply:
> The DLAF C API requires a single character, not a string.

The Fortran type for a single character is `CHARACTER` - note the absence of the `len=` argument.

The C bindings for passing a single character as value is:

```
CHARACTER(kind=C_CHAR), VALUE  :: uplo
```

Reply:
This is what I used, and did not seem work: https://github.com/RMeli/fortran-playground/blob/b84b82b7879471f90d4ba6aea36cab2b699c01ed/fortran_to_c_char/bar.F#L22-L31

Reply:
IIRC `pdpotrf` is only called with `'U'` anyways, that's why I went with this easier solution instead.

Reply:
> This is what I used, and did not seem work:

Fascinating! This feels like a bug.

I eventually got it to work by passing an integer, which should technically not be a hack.
```
   subroutine foo_char_rubbish(aa)
      character :: aa

      interface
         subroutine foo1_c(aaa) bind(C, name="foo1")
            import:: C_SIGNED_CHAR
            INTEGER(kind=C_SIGNED_CHAR), value :: aaa
         end subroutine
      end interface

      call foo1_c(IACHAR(aa, C_SIGNED_CHAR))
   end subroutine
```

Reply:
I think it's always good to remove unnecessary code. So, personally I'd remove `cp_dlaf_utils.c`. Furthermore, I'd merge `cp_dlaf_utils_api.F`  into `cp_fm_dlaf_api.F` to have that single interface module. 

However, since this affects only the DLAF code, I'll leave it up to you.

Otherwise this PR looks good to me.

Reply:
I used your workaround (thanks!) to remove the wrapper functions. I also removed the `.c` file as a whole, as requested. I'm not entirely convinced that passing `char**` directly from Fortran is better than a simple wrapper in this case, but I think it's alright.

The separation of the two `.F` is to avoid circular dependencies.